src/ontodia/data/sparql/responseHandler.ts

```diff
@@ -18,46 +18,72 @@ export function getClassTree(response: SparqlResponse<ClassBinding>): ClassModel
     const tree: ClassModel[] = [];
     const treeNodes = createClassMap(response.results.bindings);
     // createClassMap ensures we get both elements and parents and we can use treeNodes[treeNode.parent] safely
-    for (const nodeId in treeNodes) {
-        if (!treeNodes.hasOwnProperty(nodeId)) { continue; }
-        const treeNode = treeNodes[nodeId];
-        if (treeNode.parent) {
-            const parent = treeNodes[treeNode.parent];
-            parent.children.push(treeNode);
+    treeNodes.forEach(node => {
+        if (node.parents.size > 0) {
+            node.parents.forEach(parent => {
+                treeNodes.get(parent).children.push(node);
+            });
         } else {
-            tree.push(treeNode);
+            tree.push(node);
```

Comment:
Maybe I am wrong, I had no time to investigate the question, but I am not sure about the order. In some cases some node could not have a parents for a while, but actually will have them in a process of processing. If I am wrong Just skip this comment.

Reply:
I think it should be fine because `createClassMap` accounts for this and uses additional step to create nodes for parents without bindings.